### PR TITLE
Add guidelines about pair programming.

### PR DESCRIPTION
**Pull request by mathias-abtion**
This document provides an in-depth overview of pair programming and its various styles.

The aim is to inspire people to test different ways of pairing so we can figure out what works and what does not. We will eventually follow up with a survey about peoples' experiences.

### CODE REVIEW

**Comment by NorfeldtAbtion** [comment]
Nice work 👍

File: best-practices/pair-programming.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+extract this method?"
+
+The Driver needs to be aware of not taking over the pairing session, coding
+solo, which would be reminiscent of the Tour guide and Tourist style.
+
+The Navigator needs to be aware of not micro-managing, which would be
+reminiscent of the Backseat Navigator style. 
+
+This awareness is especially important when one party has more experience.
+
```
Comment (runephilosof-abtion) [inline best-practices/pair-programming.md:97]:
```suggestion

I can be beneficial to agree on a schedule for switching roles. For instance every 30 minutes.
```

**Review by runephilosof-abtion — APPROVED** [review]
Great.
Maybe add something about when to switch roles in driver and navigator.

File: best-practices/pair-programming.md (hunk 1 above)
Comment (MarkusAbtion) [reply best-practices/pair-programming.md:97]: Good idea!

Comment (MarkusAbtion) [reply best-practices/pair-programming.md:97]: We'll add a paragraph about it later today

Comment (hcarreras) [reply best-practices/pair-programming.md:97]: Agree. Maybe a good opportunity to nudge Pomodoro?

**Review by hcarreras — APPROVED** [review]
I'm loving it. I especially like the `when to use` section.
Have you thought about putting a "watch out for blindspots" section? You are already warning about some potential pitfalls, but I wonder how easy it is for other developers to add their learnings. Let's say tomorrow a pair using the backseat navigator finds a use case where this style of pairing brought a lot of value. I want them to come here and add their experience. On the other hand, if after a retro, they decide that a specific pairing style was not useful, I also want them to come here and share it with the world.

Those are just thoughts and this is perfectly good to be merged 👍 
I'll make sure that it is mentioned in the newsletter.

**Comment by MarkusAbtion** [comment]
> I'm loving it. I especially like the `when to use` section.
> Have you thought about putting a "watch out for blindspots" section? You are already warning about some potential pitfalls, but I wonder how easy it is for other developers to add their learnings. Let's say tomorrow a pair using the backseat navigator finds a use case where this style of pairing brought a lot of value. I want them to come here and add their experience. On the other hand, if after a retro, they decide that a specific pairing style was not useful, I also want them to come here and share it with the world.
> 
> Those are just thoughts and this is perfectly good to be merged 👍
> I'll make sure that it is mentioned in the newsletter.

We won't add the blindspots-section for now, but it could be an idea to "bulletize" each style more to get an easy overview.
We're currently working on a survey so we can structure the initial feedback :)

**Comment by mathias-abtion** [comment]
@prrrnd: You appear be the author of [this document](https://github.com/abtion/guidelines/blob/main/axp/README.md) which seems relevant to the one added in this pull request. Would you be up for maybe combining the two so we don't have two documents about pair programming? :-)

**Pull request merged**